tests/test_metrics.py

```diff
@@ -29,76 +29,103 @@ class Meta:
-    today = timezone.now().date()
-    # TODO: should be in settings
-    dateformat = "%Y.%m.%d"
-    today_formatted = today.strftime(dateformat)
-    assert PreprintView.get_index_name() == "osf_metrics_preprintviews-{}".format(
-        today_formatted
-    )
-
-
-@pytest.mark.es
-def test_create_metric_creates_template_with_mapping(client):
-    PreprintView.create_index_template()
-    template_name = PreprintView._template_name
-    template = client.indices.get_template(name=template_name)
-    mappings = template[template_name]["mappings"]
-    assert mappings["doc"]["_all"]["enabled"] is False
-    assert mappings["doc"]["_source"]["enabled"] is False
-    properties = mappings["doc"]["properties"]
-    assert "timestamp" in properties
-    assert properties["timestamp"] == {"doc_values": True, "type": "date"}
-    assert properties["provider_id"] == {"type": "keyword", "index": True}
-    assert properties["user_id"] == {"type": "keyword", "index": True}
-    assert properties["preprint_id"] == {"type": "keyword", "index": True}
-
-
-def test_create_metric_sends_pre_index_template_create_signal():
-    mock_listener = mock.Mock()
-    pre_index_template_create.connect(mock_listener)
-    PreprintView.create_index_template()
-    assert mock_listener.call_count == 1
-    assert "index_template" in mock_listener.call_args[1]
-
-
-# TODO: Can we make this test not use ES?
-@pytest.mark.es
-def test_source_may_be_enabled(client):
-    class MyMetric(Metric):
-        class Meta:
-            template_name = "mymetric"
-            template = "mymetric-*"
-            source = MetaField(enabled=True)
-
-    MyMetric.create_index_template()
-    template_name = MyMetric._template_name
-    template = client.indices.get_template(name=template_name)
-    mappings = template[template_name]["mappings"]
-    assert mappings["doc"]["_all"]["enabled"] is False
-    assert mappings["doc"]["_source"]["enabled"] is True
+class TestPreprintView:
+    @classmethod
+    def setup_class(cls):
+        # TODO hook into pytest.mark.es to delete indices
+        client().indices.delete(index="*")
+        client().indices.delete_template("*")
+
+    @classmethod
+    def teardown_class(cls):
+        client().indices.delete(index="*")
+        client().indices.delete_template("*")
+
+    def test_get_index_template_returns_template_with_correct_name_and_pattern(self):
+        template = PreprintView.get_index_template()
+        assert isinstance(template, IndexTemplate)
+        assert template._template_name == "osf_metrics_preprintviews"
+        assert "osf_metrics_preprintviews-*" in template.to_dict()["index_patterns"]
+
+    def test_get_index_template_respects_index_settings(self):
+        template = PreprintView.get_index_template()
+        assert template._index.to_dict()["settings"] == {"refresh_interval": "-1"}
+
+    @pytest.mark.es
+    def test_create_index(self, client):
+        PreprintView.create_index_template()
```

Comment:
Based on our previous discussion, this line should still pass without this line (because `create_index` should respect the defined mapping). Can you make that work?